### PR TITLE
Dashboard hoc

### PR DESCRIPTION
**Pull request by Ranjith-K-07**
- Navigating user to the Dashboard based on location mapping - (breadcrumb issue remaining)
- Coloring Sensor and Device based on alert type - Completed

### CODE REVIEW

File: src/components/dashboard/subComponent/deviceCard/DeviceWidget.jsx [hunk 1]
```diff
@@ -29,23 +29,23 @@ function DeviceWidget({
     });
 
     alertObject?.map((data) => {
+      if(element.alertPriority > data.alertPriority){
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/deviceCard/DeviceWidget.jsx:32]: Can this be shifted to other file ? This alert functionality should be put inside a utility or a help function file. 

Return value for this function can be consumed in this file

File: src/components/dashboard/subComponent/deviceCard/DeviceWidget.jsx [hunk 2]
```diff
@@ -29,23 +29,23 @@ function DeviceWidget({
...
+          break;
+          case 'Warning' : setAlertStatus(2);
+          break;
+          case 'outOfRange' : setAlertStatus(3);
+          break;
+          default : break;
+        }
+      } 
       element = element.alertPriority < data.alertPriority ? element
         : {
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/deviceCard/DeviceWidget.jsx:44]: Same as this one. 

Can this be shifted to other file ? This alert functionality should be put inside a utility or a help function file.

Return value for this function can be consumed in this file

File: src/components/dashboard/subComponent/landingPageComponents/MachineCard.jsx [hunk 3]
```diff
@@ -7,19 +7,66 @@ import Tooltip from '@mui/material/Tooltip';
 import { CardActionArea } from '@mui/material';
 import Stack from '@mui/material/Stack';
 import MachineCircularProgressbar from './MachineCircularProgressbar';
+import { getSensorBackgroundColor, getSensorHeaderColor } from '../../../../utils/helperFunctions';
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/landingPageComponents/MachineCard.jsx:10]: THis is the link of helper function I'm referring to

File: src/components/dashboard/subComponent/landingPageComponents/MachineCard.jsx [hunk 4]
```diff
@@ -7,19 +7,66 @@ import Tooltip from '@mui/material/Tooltip';
...
+      alertLabel: 'Good',
+      alertColor: 'green',
+      alertPriority: 4,
+    };
+
+    const alertObject = props.sensorIdList?.filter((alert) => {
+      return parseInt(props.id) === parseInt(alert.id);
+    });
+    
+    alertObject?.map((data) => {
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/landingPageComponents/MachineCard.jsx:27]: This code if pushed to helper function then entire code will be reused from helper function

File: src/components/dashboard/subComponent/landingPageComponents/MachineCard.jsx [hunk 5]
```diff
@@ -7,19 +7,66 @@ import Tooltip from '@mui/material/Tooltip';
...
+          case 'Critical' : setAlertStatus(1);
+          break;
+          case 'Warning' : setAlertStatus(2);
+          break;
+          case 'outOfRange' : setAlertStatus(3);
+          break;
+          default : break;
+        }
+      } 
+      element = element.alertPriority < data.alertPriority ? element
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/landingPageComponents/MachineCard.jsx:39]: Same as above

File: src/components/dashboard/subComponent/siteDetailsComponent/BranchGridComponent.js [hunk 6]
```diff
@@ -46,9 +46,9 @@ function BranchGridComponent({
         alertObject?.map((data) => {
           element = element.alertPriority < data.alertPriority ? element
             : {
-              alertLabel: data.alertType === 'Critical' ? 'Critical' : data.alertType === 'outOfRange' ? 'Out Of Range' : 'Good',
-              alertColor: data.alertType === 'Critical' ? 'red' : data.alertType === 'outOfRange' ? 'orange' : 'green',
-              alertPriority: data.alertType === 'Critical' ? 1 : data.alertType === 'outOfRange' ? 2 : 3,
+              alertLabel: data.alertType === 'Critical'? 'Critical' : data.alertType === 'outOfRange'? 'Out Of Range' : data.alertType === 'Warning' ? 'Warning' :'Good',
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/siteDetailsComponent/BranchGridComponent.js:49]: Reuse this code from the above helper function after resolving the above comments

File: src/components/dashboard/subComponent/siteDetailsComponent/BuildingGridComponent.js [hunk 7]
```diff
@@ -44,9 +44,9 @@ function BuildingGridComponent({
         alertObject?.map((data) => {
           element = element.alertPriority < data.alertPriority ? element
             : {
-              alertLabel: data.alertType === 'Critical' ? 'Critical' : data.alertType === 'outOfRange' ? 'Out Of Range' : 'Good',
-              alertColor: data.alertType === 'Critical' ? 'red' : data.alertType === 'outOfRange' ? 'orange' : 'green',
-              alertPriority: data.alertType === 'Critical' ? 1 : data.alertType === 'outOfRange' ? 2 : 3,
+              alertLabel: data.alertType === 'Critical'? 'Critical' : data.alertType === 'outOfRange'? 'Out Of Range' : data.alertType === 'Warning' ? 'Warning' :'Good',
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/siteDetailsComponent/BuildingGridComponent.js:47]: Reuse this code

File: src/components/dashboard/subComponent/siteDetailsComponent/FacilityGridComponent.js [hunk 8]
```diff
@@ -44,9 +44,9 @@ function FacilityGridComponent({
         alertObject?.map((data) => {
           element = element.alertPriority < data.alertPriority ? element
             : {
-              alertLabel: data.alertType === 'Critical' ? 'Critical' : data.alertType === 'outOfRange' ? 'Out Of Range' : 'Good',
-              alertColor: data.alertType === 'Critical' ? 'red' : data.alertType === 'outOfRange' ? 'orange' : 'green',
-              alertPriority: data.alertType === 'Critical' ? 1 : data.alertType === 'outOfRange' ? 2 : 3,
+              alertLabel: data.alertType === 'Critical'? 'Critical' : data.alertType === 'outOfRange'? 'Out Of Range' : data.alertType === 'Warning' ? 'Warning' :'Good',
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/siteDetailsComponent/FacilityGridComponent.js:47]: reuse this code

File: src/components/dashboard/subComponent/siteDetailsComponent/FloorGridComponent.js [hunk 9]
```diff
@@ -45,9 +45,9 @@ function FloorGridComponent({
         alertObject?.map((data) => {
           element = element.alertPriority < data.alertPriority ? element
             : {
-              alertLabel: data.alertType === 'Critical' ? 'Critical' : data.alertType === 'outOfRange' ? 'Out Of Range' : 'Good',
-              alertColor: data.alertType === 'Critical' ? 'red' : data.alertType === 'outOfRange' ? 'orange' : 'green',
-              alertPriority: data.alertType === 'Critical' ? 1 : data.alertType === 'outOfRange' ? 2 : 3,
+              alertLabel: data.alertType === 'Critical'? 'Critical' : data.alertType === 'outOfRange'? 'Out Of Range' : data.alertType === 'Warning' ? 'Warning' :'Good',
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/siteDetailsComponent/FloorGridComponent.js:48]: reuse this code from helper function

File: src/components/dashboard/subComponent/siteDetailsComponent/LabGridComponent.js [hunk 10]
```diff
@@ -44,9 +44,9 @@ function LabGridComponent({
         alertObject?.map((data) => {
           element = element.alertPriority < data.alertPriority ? element
             : {
-              alertLabel: data.alertType === 'Critical' ? 'Critical' : data.alertType === 'outOfRange' ? 'Out Of Range' : 'Good',
-              alertColor: data.alertType === 'Critical' ? 'red' : data.alertType === 'outOfRange' ? 'orange' : 'green',
-              alertPriority: data.alertType === 'Critical' ? 1 : data.alertType === 'outOfRange' ? 2 : 3,
+              alertLabel: data.alertType === 'Critical'? 'Critical' : data.alertType === 'outOfRange'? 'Out Of Range' : data.alertType === 'Warning' ? 'Warning' :'Good',
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/siteDetailsComponent/LabGridComponent.js:47]: this too

File: src/components/dashboard/subComponent/siteDetailsComponent/LocationGridComponent.js [hunk 11]
```diff
@@ -44,8 +44,8 @@ function LocationGridComponent(props) {
         alertObject?.map((data)=>{
           element = element.alertPriority < data.alertPriority ? element : 
             { 
-              alertLabel: data.alertType === 'Critical'? 'Critical' : data.alertType === 'outOfRange'? 'Out Of Range' : 'Good',
-              alertColor : data.alertType === 'Critical'? 'red' : data.alertType === 'outOfRange'? 'orange' : data.alertType === 'Warning' ? '#9c27b0' : 'green',
+              alertLabel: data.alertType === 'Critical'? 'Critical' : data.alertType === 'outOfRange'? 'Out Of Range' : data.alertType === 'Warning' ? 'Warning' :'Good',
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/siteDetailsComponent/LocationGridComponent.js:47]: reuse this code

File: src/utils/helperFunctions.js [hunk 12]
```diff
@@ -111,4 +132,41 @@ export function getDeviceModeColor(deviceMode){
...
 }
-// 1b5e20
\ No newline at end of file
+
+export function getSensorBackgroundColor(sensorStatus, alertStatus){
+  let colorCode = '#a5d6a7';
+    if(sensorStatus === '0') {
+      colorCode = '#9e9e9e';
+    } else {
+        switch(alertStatus){
```
Comment (teamlead-rdl) [inline src/utils/helperFunctions.js:141]: substitute this switch at a single function above in this same file and resuse this SWTICH CASE to other functions

File: src/utils/helperFunctions.js [hunk 13]
```diff
@@ -111,4 +132,41 @@ export function getDeviceModeColor(deviceMode){
...
+  return colorCode;
+}
+
+export function getSensorHeaderColor(sensorStatus, alertStatus){
+  let colorCode = '#212121';
+    if(sensorStatus === '0') {
+      colorCode = '#212121';
+    } else {
+        switch(alertStatus){
+          case 1 : colorCode = '#b71c1c';
```
Comment (teamlead-rdl) [inline src/utils/helperFunctions.js:161]: substitute this switch at a single function above in this same file and resuse this SWTICH CASE to other functions

File: src/utils/helperFunctions.js [hunk 14]
```diff
@@ -85,16 +84,38 @@ export function getDeviceHeaderColor(deviceMode, alertStatus){
...
+export function getDeviceBackgroundColor(deviceMode, alertStatus){
+  let colorCode = '#a5d6a7';
+    if(deviceMode === 'disabled') {
+      colorCode = '#9e9e9e';
+    } else {
+      if(deviceMode === 'bumpTest' || deviceMode === 'calibration' || deviceMode === 'firmwareUpgradation' || deviceMode === 'config'){
+        colorCode = '#f8bbd0';
+      } else if(deviceMode === 'enabled'){
+        switch(alertStatus){
+          case 1 : colorCode = '#ef9a9a';
```
Comment (teamlead-rdl) [inline src/utils/helperFunctions.js:107]: substitute this switch at a single function above in this same file and resuse this SWTICH CASE to other functions

File: src/utils/helperFunctions.js [hunk 15]
```diff
@@ -85,16 +84,38 @@ export function getDeviceHeaderColor(deviceMode, alertStatus){
       switch(alertStatus){
         case 1 : colorCode = '#b71c1c';
```
Comment (teamlead-rdl) [inline src/utils/helperFunctions.js:85]: substitute this switch at a single function above in this same file and resuse this SWTICH CASE to other functions

File: src/utils/helperFunctions.js [hunk 16]
```diff
@@ -61,22 +61,21 @@ export function getDeviceBackgroundColor(deviceMode, alertStatus){
         switch(alertStatus){
           case 1 : colorCode = '#ef9a9a';
```
Comment (teamlead-rdl) [inline src/utils/helperFunctions.js:62]: substitute this switch at a single function above in this same file and resuse this SWTICH CASE to other functions

File: src/components/dashboard/subComponent/deviceCard/DeviceWidget.jsx (hunk 1 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/deviceCard/DeviceWidget.jsx:32]: Changes taken care

File: src/components/dashboard/subComponent/deviceCard/DeviceWidget.jsx (hunk 2 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/deviceCard/DeviceWidget.jsx:44]: Changes taken care

File: src/components/dashboard/subComponent/landingPageComponents/MachineCard.jsx (hunk 5 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/landingPageComponents/MachineCard.jsx:39]: Changes taken care

File: src/components/dashboard/subComponent/siteDetailsComponent/BuildingGridComponent.js (hunk 7 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/siteDetailsComponent/BuildingGridComponent.js:47]: Changes taken care

File: src/components/dashboard/subComponent/siteDetailsComponent/FacilityGridComponent.js (hunk 8 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/siteDetailsComponent/FacilityGridComponent.js:47]: Changes taken care

File: src/components/dashboard/subComponent/siteDetailsComponent/FloorGridComponent.js (hunk 9 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/siteDetailsComponent/FloorGridComponent.js:48]: Changes taken care

File: src/components/dashboard/subComponent/siteDetailsComponent/LabGridComponent.js (hunk 10 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/siteDetailsComponent/LabGridComponent.js:47]: Changes taken care

File: src/components/dashboard/subComponent/siteDetailsComponent/LocationGridComponent.js (hunk 11 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/siteDetailsComponent/LocationGridComponent.js:47]: Changes taken care

File: src/utils/helperFunctions.js (hunk 12 above)
Comment (Ranjith-K-07) [reply src/utils/helperFunctions.js:141]: Changes taken care

File: src/utils/helperFunctions.js (hunk 13 above)
Comment (Ranjith-K-07) [reply src/utils/helperFunctions.js:161]: Changes taken care

File: src/utils/helperFunctions.js (hunk 15 above)
Comment (Ranjith-K-07) [reply src/utils/helperFunctions.js:85]: Changes taken care

File: src/utils/helperFunctions.js (hunk 16 above)
Comment (Ranjith-K-07) [reply src/utils/helperFunctions.js:62]: Changes taken care

File: src/utils/helperFunctions.js (hunk 14 above)
Comment (Ranjith-K-07) [reply src/utils/helperFunctions.js:107]: Changes taken care

File: src/components/dashboard/subComponent/deviceCard/DeviceWidget.jsx [hunk 17]
```diff
@@ -29,23 +29,25 @@ function DeviceWidget({
...
+      //     break;
+      //     case 'Warning' : setAlertStatus(2);
+      //     break;
+      //     case 'outOfRange' : setAlertStatus(3);
+      //     break;
+      //     default : break;
+      //   }
+      // } 
+      element = setAlertPriorityAndType(element, data);
+      // element = element.alertPriority < data.alertPriority ? element
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/deviceCard/DeviceWidget.jsx:45]: Please remove the commented code.

File: src/components/dashboard/subComponent/siteDetailsComponent/BranchGridComponent.js [hunk 18]
```diff
@@ -37,19 +38,20 @@ function BranchGridComponent({
...
 
         alertObject?.map((data) => {
-          element = element.alertPriority < data.alertPriority ? element
-            : {
-              alertLabel: data.alertType === 'Critical' ? 'Critical' : data.alertType === 'outOfRange' ? 'Out Of Range' : 'Good',
-              alertColor: data.alertType === 'Critical' ? 'red' : data.alertType === 'outOfRange' ? 'orange' : 'green',
-              alertPriority: data.alertType === 'Critical' ? 1 : data.alertType === 'outOfRange' ? 2 : 3,
-            };
+          element = setAlertPriorityAndType(element, data);
+          // element = element.alertPriority < data.alertPriority ? element
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/siteDetailsComponent/BranchGridComponent.js:49]: Please remove the commented code.

File: src/components/dashboard/subComponent/siteDetailsComponent/BuildingGridComponent.js [hunk 19]
```diff
@@ -35,19 +36,20 @@ function BuildingGridComponent({
...
 
         alertObject?.map((data) => {
-          element = element.alertPriority < data.alertPriority ? element
-            : {
-              alertLabel: data.alertType === 'Critical' ? 'Critical' : data.alertType === 'outOfRange' ? 'Out Of Range' : 'Good',
-              alertColor: data.alertType === 'Critical' ? 'red' : data.alertType === 'outOfRange' ? 'orange' : 'green',
-              alertPriority: data.alertType === 'Critical' ? 1 : data.alertType === 'outOfRange' ? 2 : 3,
-            };
+          element = setAlertPriorityAndType(element, data);
+          // element = element.alertPriority < data.alertPriority ? element
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/siteDetailsComponent/BuildingGridComponent.js:47]: Please remove the commented code.

File: src/components/dashboard/subComponent/siteDetailsComponent/FacilityGridComponent.js [hunk 20]
```diff
@@ -35,19 +36,20 @@ function FacilityGridComponent({
...
 
         alertObject?.map((data) => {
-          element = element.alertPriority < data.alertPriority ? element
-            : {
-              alertLabel: data.alertType === 'Critical' ? 'Critical' : data.alertType === 'outOfRange' ? 'Out Of Range' : 'Good',
-              alertColor: data.alertType === 'Critical' ? 'red' : data.alertType === 'outOfRange' ? 'orange' : 'green',
-              alertPriority: data.alertType === 'Critical' ? 1 : data.alertType === 'outOfRange' ? 2 : 3,
-            };
+          element = setAlertPriorityAndType(element, data);
+          // element = element.alertPriority < data.alertPriority ? element
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/siteDetailsComponent/FacilityGridComponent.js:47]: Please remove the commented code.

File: src/components/dashboard/subComponent/siteDetailsComponent/FloorGridComponent.js [hunk 21]
```diff
@@ -36,19 +37,20 @@ function FloorGridComponent({
...
 
         alertObject?.map((data) => {
-          element = element.alertPriority < data.alertPriority ? element
-            : {
-              alertLabel: data.alertType === 'Critical' ? 'Critical' : data.alertType === 'outOfRange' ? 'Out Of Range' : 'Good',
-              alertColor: data.alertType === 'Critical' ? 'red' : data.alertType === 'outOfRange' ? 'orange' : 'green',
-              alertPriority: data.alertType === 'Critical' ? 1 : data.alertType === 'outOfRange' ? 2 : 3,
-            };
+          element = setAlertPriorityAndType(element, data);
+          // element = element.alertPriority < data.alertPriority ? element
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/siteDetailsComponent/FloorGridComponent.js:48]: Please remove the commented code.

File: src/components/dashboard/subComponent/siteDetailsComponent/LabGridComponent.js [hunk 22]
```diff
@@ -35,19 +36,20 @@ function LabGridComponent({
...
 
         alertObject?.map((data) => {
-          element = element.alertPriority < data.alertPriority ? element
-            : {
-              alertLabel: data.alertType === 'Critical' ? 'Critical' : data.alertType === 'outOfRange' ? 'Out Of Range' : 'Good',
-              alertColor: data.alertType === 'Critical' ? 'red' : data.alertType === 'outOfRange' ? 'orange' : 'green',
-              alertPriority: data.alertType === 'Critical' ? 1 : data.alertType === 'outOfRange' ? 2 : 3,
-            };
+          element = setAlertPriorityAndType(element, data);
+          // element = element.alertPriority < data.alertPriority ? element
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/siteDetailsComponent/LabGridComponent.js:47]: Please remove the commented code.

File: src/components/dashboard/subComponent/siteDetailsComponent/LocationGridComponent.js [hunk 23]
```diff
@@ -42,12 +43,13 @@ function LocationGridComponent(props) {
...
 
         alertObject?.map((data)=>{
-          element = element.alertPriority < data.alertPriority ? element : 
-            { 
-              alertLabel: data.alertType === 'Critical'? 'Critical' : data.alertType === 'outOfRange'? 'Out Of Range' : 'Good',
-              alertColor : data.alertType === 'Critical'? 'red' : data.alertType === 'outOfRange'? 'orange' : data.alertType === 'Warning' ? '#9c27b0' : 'green',
-              alertPriority: data.alertType === 'Critical'? 1 : data.alertType === 'outOfRange'? 2 : data.alertType === 'Warning' ? 3 : 4
-            }
+          element = setAlertPriorityAndType(element, data);
+          // element = element.alertPriority < data.alertPriority ? element :
```
Comment (teamlead-rdl) [inline src/components/dashboard/subComponent/siteDetailsComponent/LocationGridComponent.js:47]: Please remove the commented code.

File: src/utils/helperFunctions.js [hunk 24]
```diff
@@ -50,49 +65,49 @@ export function setAlertColor(newNotificationStack){
...
-        case 2 : colorCode = '#ffb74d';
-          break;
-        case 3 : colorCode = '#ef6c00';
-          break;
-        case 4 : colorCode = '#1b5e20';
-          break;
-        default : break;
-      }
+      colorCode = handleSwitchcase(alertStatus, colorCode, '#1b5e20', '#b71c1c', '#e65100', '#4a148c');
+      // switch(alertStatus){
```
Comment (teamlead-rdl) [inline src/utils/helperFunctions.js:101]: Please remove the commented code.

File: src/utils/helperFunctions.js [hunk 25]
```diff
@@ -111,4 +126,66 @@ export function getDeviceModeColor(deviceMode){
...
-// 1b5e20
\ No newline at end of file
+
+export function getSensorBackgroundColor(sensorStatus, alertStatus){
+  let colorCode = '#a5d6a7';
+    if(sensorStatus === '0') {
+      colorCode = '#9e9e9e';
+    } else {
+      colorCode = handleSwitchcase(alertStatus, colorCode, '#a5d6a7', '#ef9a9a', '#ffb74d', '#ce93d8');
+        // switch(alertStatus){
```
Comment (teamlead-rdl) [inline src/utils/helperFunctions.js:136]: Please remove the commented code.

File: src/utils/helperFunctions.js [hunk 26]
```diff
@@ -50,49 +65,49 @@ export function setAlertColor(newNotificationStack){
...
-          case 2 : colorCode = '#7e57c2';
-            break;
-          case 3 : colorCode = '#ffb74d';
-            break;
-          case 4 : colorCode = '#a5d6a7';
-            break;
-          default : break;
-        }
+        colorCode = handleSwitchcase(alertStatus, colorCode, '#a5d6a7', '#ef9a9a', '#ffb74d', '#ce93d8');
+        // switch(alertStatus){
```
Comment (teamlead-rdl) [inline src/utils/helperFunctions.js:77]: Please remove the commented code.

File: src/components/dashboard/subComponent/deviceCard/DeviceWidget.jsx (hunk 17 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/deviceCard/DeviceWidget.jsx]: Taken care

File: src/components/dashboard/subComponent/siteDetailsComponent/BranchGridComponent.js (hunk 18 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/siteDetailsComponent/BranchGridComponent.js]: Taken care

File: src/components/dashboard/subComponent/siteDetailsComponent/BuildingGridComponent.js (hunk 19 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/siteDetailsComponent/BuildingGridComponent.js]: Taken care

File: src/components/dashboard/subComponent/siteDetailsComponent/FacilityGridComponent.js (hunk 20 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/siteDetailsComponent/FacilityGridComponent.js]: Taken care

File: src/components/dashboard/subComponent/siteDetailsComponent/FloorGridComponent.js (hunk 21 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/siteDetailsComponent/FloorGridComponent.js]: Taken care

File: src/components/dashboard/subComponent/siteDetailsComponent/LabGridComponent.js (hunk 22 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/siteDetailsComponent/LabGridComponent.js]: Taken care

File: src/components/dashboard/subComponent/siteDetailsComponent/LocationGridComponent.js (hunk 23 above)
Comment (Ranjith-K-07) [reply src/components/dashboard/subComponent/siteDetailsComponent/LocationGridComponent.js]: Taken care

File: src/utils/helperFunctions.js (hunk 24 above)
Comment (Ranjith-K-07) [reply src/utils/helperFunctions.js]: Taken care

File: src/utils/helperFunctions.js (hunk 25 above)
Comment (Ranjith-K-07) [reply src/utils/helperFunctions.js]: Taken care

File: src/utils/helperFunctions.js (hunk 26 above)
Comment (Ranjith-K-07) [reply src/utils/helperFunctions.js]: Taken care

**Pull request merged**